pkg/controller/scylladbcluster/sync.go

```diff
@@ -266,108 +256,163 @@ func (scc *Controller) sync(ctx context.Context, key string) error {
 
 	var errs []error
 
-	err = controllerhelpers.RunSync(
-		&status.Conditions,
-		remoteNamespaceControllerProgressingCondition,
-		remoteNamespaceControllerDegradedCondition,
-		sc.Generation,
-		func() ([]metav1.Condition, error) {
-			return scc.syncNamespaces(ctx, sc, remoteNamespaceMap, managingClusterDomain)
-		},
-	)
-	if err != nil {
-		errs = append(errs, fmt.Errorf("can't sync remote namespaces: %w", err))
-	}
+	for _, dc := range sc.Spec.Datacenters {
+		objectErrs := objectErrMaps[dc.RemoteKubernetesClusterName]
 
-	err = controllerhelpers.RunSync(
-		&status.Conditions,
-		remoteRemoteOwnerControllerProgressingCondition,
-		remoteRemoteOwnerControllerDegradedCondition,
-		sc.Generation,
-		func() ([]metav1.Condition, error) {
-			return scc.syncRemoteOwners(ctx, sc, remoteNamespaces, remoteRemoteOwnerMap, managingClusterDomain)
-		},
-	)
-	if err != nil {
-		errs = append(errs, fmt.Errorf("can't sync remote remoteowners: %w", err))
-	}
+		var err error
+		err = utilerrors.NewAggregate(objectErrs)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("can't sync remote datacenter %q: %w", dc.RemoteKubernetesClusterName, err))
+			continue
+		}
 
-	err = controllerhelpers.RunSync(
-		&status.Conditions,
-		remoteServiceControllerProgressingCondition,
-		remoteServiceControllerDegradedCondition,
-		sc.Generation,
-		func() ([]metav1.Condition, error) {
-			return scc.syncServices(ctx, sc, remoteNamespaces, remoteControllers, remoteServiceMap, managingClusterDomain)
-		},
-	)
-	if err != nil {
-		errs = append(errs, fmt.Errorf("can't sync remote services: %w", err))
-	}
+		err = controllerhelpers.RunSync(
+			&status.Conditions,
+			fmt.Sprintf(remoteNamespaceControllerDatacenterProgressingConditionFormat, dc.Name),
```

Comment:
There seems to be an existing pattern in package `naming`. Is there a reason why this sprintf would be better than

```go
naming.RemoteSyncCondition("Namespace", scyllav1alpha1.ProgressingCondition, dc.Name)
```

?

(You may have a better idea for the library function name)

Reply:
The convention of keeping the conditions enumerated in `conditions.go` for consistency across controllers is okay. The comment is about why not apply a similar convention of using the package `naming` for well-known names that are part of the public interface.

The reason is beneficial is expressed in my other comment -- that then these names could be "excluded" from the system under test in the E2Es to make them duplicate the logic less.